### PR TITLE
[3869] Close form-based views when the underlying element no longer exists

### LINKED ISSUE #3869: The properties of a deleted object are still available in the details view
* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- 2024.7.10
* [x] **I have specified my environment** : 
- All

### Steps to reproduce

1/ Create a project "Sirius Web" from the template.
2/ Select any object form the Explorer view that has properties in the Details view
3/ Delete the object

### Expected behavior

The properties of the deleted object are not visible anymore in the Details view

### Actual behavior

The properties of the deleted object are still visible and editable in the Details view, which can lead to errors/crashes.

### PR DESCRIPTION
**Pull request by pcdavid**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3869
Signed-off-by: Pierre-Charles David <pierre-charles.david@obeo.fr>

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/sirius-web/frontend/sirius-web-application/src/diagrams/useDiagramFilterSubscription.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const useDiagramFilterSubscription = (
       const { diagramFilterEvent: payload } = gqlDiagramFilterEventSubscription;
       if (isFormRefreshedEventPayload(payload)) {
         const { form } = payload;
-        setState((prevState) => ({ ...prevState, form }));
+        setState((prevState) => ({ ...prevState, form, complete: false }));
```
Comment (sbegaudeau) [inline packages/sirius-web/frontend/sirius-web-application/src/diagrams/useDiagramFilterSubscription.tsx:68]: The `...prevState` should be enough isn't it? Same comment for all the other hooks.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: packages/sirius-web/frontend/sirius-web-application/src/diagrams/useDiagramFilterSubscription.tsx (hunk 1 above)
Comment (pcdavid) [reply packages/sirius-web/frontend/sirius-web-application/src/diagrams/useDiagramFilterSubscription.tsx:68]: That was my first attempt, but without this explicit `complete: false` the Details view is always empty. It seems we receive an `onComplete` invocation at some point and then if we don't reset `complete` to false it stays `true`.

Comment (pcdavid) [reply packages/sirius-web/frontend/sirius-web-application/src/diagrams/useDiagramFilterSubscription.tsx:68]: The 'onComplete' call happens when I select a Document in the explorer.
When interactive with the explorer, the only way to find a proper element to select involves selecting a document as the first step, which does not produce a details view, hence the onComplete.

If I navigate to a diagram's page directly, and then select an element from the diagram, the Details view correclyt appears (without the explicit `complete: false`), but as soon as I re-select a Document, I'm back to a 'complete: true' state and without an explicit reset the view stays empty.

Comment (pcdavid) [reply packages/sirius-web/frontend/sirius-web-application/src/diagrams/useDiagramFilterSubscription.tsx:68]: Without the explicit `complete: false`:

https://github.com/user-attachments/assets/eb60b68a-107d-42e6-b15f-d088d1d1bf85

**Review by sbegaudeau — APPROVED** [review]

**Comment by sbegaudeau** [comment]
I'll rebase and merge it

**Pull request merged**